types/better-sqlite3/types.d.ts

```diff
@@ -0,0 +1,80 @@
+import Integer = require("integer");
+
+export interface RunResult {
```

Comment:
Why are you exporting all of these types if the only one that will be used is `DatabaseConstructor`? More generally, what was your intent in moving all of these to a separate file? Is it just for organization purposes?

Reply:
Two things. 

* Code organization
* Exposing some types than can be useful for devs

Reply:
Also the linter gave me the advice to do it. And for my side I need `RunResult` exported from the module.

Reply:
I see. Rather than moving everything to a separate file, I'd recommend the following.

Rename the classes `Statement` and `Transaction` to `DatabaseStatement` and `DatabaseTransaction` (or something similar, these names will not be available to consumers). Add a namespace as follows and put all of the interfaces you want to expose in it. Also, add interfaces exposing the types from Statement and Transaction as shown.
```ts
declare namespace Database {
  interface RunResult {
    ...
  }
  interface Statement extends DatabaseStatement { }
  interface Transaction extends DatabaseTransaction { }
}
```
Note that the namespace must be named `Database`. This will then allow consumers to do the following:
```ts
import Database = require("better-sqlite3");
const error = new Database.SqliteError("m", "m");
const int = Database.Integer('3');
const db = Database("");
const stmt: Database.Statement = db.prepare("");
const result: Database.RunResult = stmt.run();
```

The extra file isn't a good approach especially for the classes since users might try to import `Statement` from the other file and construct it, which would fail at runtime.

Reply:
Well, doing this makes the linter break:

![image](https://user-images.githubusercontent.com/7959437/49049311-856d4580-f1ac-11e8-8676-4253f1261dde.png)

Reply:
Maybe taking the whole interfaces into the new namespace would be a better approach.